### DISCUSSION TITLE
suggestion @Client:  allow to access inner properties of objects

### DISCUSSION THREAD

**Discussion by kbachl** [micronaut-projects/micronaut-core#4034] [Ideas]
Hi,

currently this wont work:

```
@Patch("/DeliveryNotes({note.docEntry})")
HttpResponse<Void> patchDeliveryNoteShort(@Body PartialDeliveryNotePatch note);
```

leads to: 
`error: The route declares a uri variable named [note.docEntry], but no corresponding method argument is present`

instead this is needed:

```
@Patch("/DeliveryNotes({docEntry})")
HttpResponse<Void> patchDeliveryNoteShort(int docEntry, @Body PartialDeliveryNotePatch note);
```

so I suggest to let it access inner properties of objects as it leads to smaller and cleaner code, especially when working with OData services/ web services that need a part of the body also in their path

**Comment by graemerocher** [comment]
I'm not sure this is doable, the URI variables are place holders for URI variables that have a different encoding to JSON and are used to match the request in the most efficient way possible. It would add significant performance overhead if you request matching had to start parsing the request body.

I am moving this to a discussion of ideas for the moment.

Thoughts @jameskleeh ?

**Reply by kbachl** [reply]
I see, but the error comes from compile time, not runtime... so I dont see the performance problem here as its at compile only;

**Reply by graemerocher** [reply]
it comes from compile time because you are using the framework in a currently unsupported way. If we were to add support for this feature then the overhead would be at runtime

**Reply by kbachl** [reply]
Ah, ok - now I understand. Thx!
